tsconfig.eslint.json

```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "types": ["@types/node", "jest"],
+    "noEmit": true,
+    "allowJs": true
+  },
+  "extends": "./tsconfig.json",
+  "include": [".eslintrc.js", "**/*.js", "**/__tests__/**/*.ts"]
```

Comment:
This new `tsconfig` file is used to add linting coverage to all of our non-package files: `js` configs + scripts and our test files. It just extends from the root `tsconfig.json` and overrides a few compiler options.